cli/internal/run/run.go

```diff
@@ -484,6 +506,49 @@ func noopPersistentOptsDuringMigration(flags *pflag.FlagSet) {
 	}
 }
 
+const (
+	_graphText      = "graph"
+	_graphNoValue   = "<output filename>"
+	_graphTextValue = "true"
+)
+
+// graphValue implements a flag that can be treated as a boolean (--graph)
+// or a string (--graph=output.svg).
+type graphValue struct {
+	opts *runOpts
+}
+
+var _ pflag.Value = &graphValue{}
+
+func (d *graphValue) String() string {
+	if d.opts.graphDot {
+		return _graphText
+	}
+	return d.opts.graphFile
+}
+
+func (d *graphValue) Set(value string) error {
+	if value == _graphNoValue {
```

Comment:
This feels like an unfortunate hack because of `NoOptDefVal` appearing as a string in the usage text. Is it possible that we can use `pflag.Flagset.Changed` instead of this? (I'd much prefer for this to be some sort of sigil value that a user is incapable of inputting.)

(I haven't yet looked into how pflag works so I don't know what is available when.)